### PR TITLE
FACES-229 Button image resource not rendering with Pluto

### PR DESCRIPTION
**Pull request by stiemannkj1**

### CODE REVIEW

File: bridge-impl/src/main/java/com/liferay/faces/bridge/internal/BridgeURLResourceImpl.java [hunk 1]
```diff
@@ -49,15 +51,15 @@
 	private static final Logger logger = LoggerFactory.getLogger(BridgeURLResourceImpl.class);
 
 	// Private Constants
-	private static final String ENCODED_RESOURCE_TOKEN = "javax.faces.resource=";
-
-	// Private Pseudo-Constants
-	private static Set<String> EXCLUDED_PARAMETER_NAMES = new HashSet<String>();
+	private static final Set<String> EXCLUDED_PARAMETER_NAMES;
```
Comment (stiemannkj1) [inline bridge-impl/src/main/java/com/liferay/faces/bridge/internal/BridgeURLResourceImpl.java:54]: @ngriffin7a, just performing some minor unrelated cleanup here.

File: bridge-impl/src/main/java/com/liferay/faces/bridge/application/internal/ViewHandlerImpl.java [hunk 2]
```diff
@@ -125,6 +126,17 @@ public String getActionURL(FacesContext facesContext, String viewId) {
 	}
 
 	@Override
+	public String getResourceURL(FacesContext fc, String string) {
+
+		if (PortletResourceUtil.isPortletResourceURL(string)) {
```
Comment (stiemannkj1) [inline bridge-impl/src/main/java/com/liferay/faces/bridge/application/internal/ViewHandlerImpl.java:131]: @ngriffin7a, I believe this change is valid because it is essentially the same change that was made inside Mojarra's `RenderKitUtils` from [JSF 2.1](https://github.com/javaserverfaces/mojarra/blob/2.1.29-05/jsf-ri/src/main/java/com/sun/faces/renderkit/RenderKitUtils.java#L1356-L1357) to [2.2](https://github.com/javaserverfaces/mojarra/blob/2.2.13/jsf-ri/src/main/java/com/sun/faces/renderkit/RenderKitUtils.java#L1403-L1404):

JSF 2.1:

```
// public static final String RESOURCE_IDENTIFIER = "/javax.faces.resource";
if (value.contains(ResourceHandler.RESOURCE_IDENTIFIER)) {
    return value;
```

JSF 2.2

```
if (handler.isResourceURL(value)) {
    return value;
```

The change in Mojarra made things work correctly in JSF 2.2 since we override `ResourceHandler.isResourceURL()`:

```
@Override
public boolean isResourceURL(String url) {
    return (url != null) && url.contains("javax.faces.resource=");
}
```

However, things worked incorrectly in JSF 2.1 because Mojarra was checking for the servlet indicator of a resource URL token.

**Comment by stiemannkj1** [comment]
@ngriffin7a, I tested this change in the following demos and environments and it passed all tests:

jsf-applicant:
- Pluto 2 + JSF 2.1
- Pluto 2 + JSF 2.2
- Liferay 6.2 + JSF 2.1

jsf-showcase:
- Pluto 2 + JSF 2.1
- Pluto 2 + JSF 2.2

primefaces-applicant:
- Pluto 2 + JSF 2.2

TCK:
- Pluto 2 + JSF 2.2
- Pluto 2 + JSF 2.1 (with the `javax.faces.DEFAULT_SUFFIX` `context-param` commented out)
- Liferay 6.2 + JSF 2.2

**Comment by stiemannkj1** [comment]
@ngriffin7a, this change can and should be backported at least to `3.x` and possibly to `2.x`.

**Comment by ngriffin7a** [comment]
Merged and backported. Thank you. Please see follow-up commit that moves the ViewHandlerImpl .java fix to ViewHandlerCompatImpl.java in the liferay-faces-bridge-impl-3.x branch only: 9e9652d3e24139ab639fcfe04e439383509fa671
